src/cloud-cli/meltano/cloud/api/client.py

```diff
@@ -15,6 +15,11 @@
 from meltano.cloud.api.auth import MeltanoCloudAuth
 from meltano.cloud.api.config import MeltanoCloudConfig
 
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
```

Comment:
This can go in a `t.TYPE_CHECKING` block afaict

Reply:
Resolved in 4befccd. Thanks!